src/test/java/spoon/test/generics/GenericsTest.java

```diff
@@ -695,36 +725,39 @@ public void testCtSuperTypeHierarchy() throws Exception {
 		
 		//type of first parameter of  method WeddingLunch#eatMe
 		CtTypeReference<?> ctWeddingLunch_X = tWeddingLunch_eatMe.getParameters().get(0).getType();
+		// X is the type parameter of WeddingLunch
 		assertEquals("X", ctWeddingLunch_X.getSimpleName());
 		//type of first parameter of method Lunch#eatMe
 		CtTypeReference<?> ctClassLunch_A = ctClassLunch_eatMe.getParameters().get(0).getType();
 		assertEquals("A", ctClassLunch_A.getSimpleName());
 		
 		//are these two types same?
-		CtSuperTypeHierarchy sthOftrWeddingLunch_Mole = new CtSuperTypeHierarchy(trWeddingLunch_Mole);
+		CtSuperTypeHierarchy typingContextOfDisgust = new CtSuperTypeHierarchy(typeReferenceOfDisgust);
 		
-		//adapt X to scope of CelebrationLunch<Integer,Long,Double>.WeddingLunch<Mole>
-		CtTypeReference<?> adapted = sthOftrWeddingLunch_Mole.adaptType(ctWeddingLunch_X);
-		assertEquals("spoon.test.generics.testclasses.Mole", adapted.getQualifiedName());
+		// in disgust, X of WeddingLunch is bound to "Model"
+		assertEquals("spoon.test.generics.testclasses.Mole", typingContextOfDisgust.adaptType(ctWeddingLunch_X).getQualifiedName());
 		//adapt A to scope of CelebrationLunch<Integer,Long,Double>.WeddingLunch<Mole>
-		adapted = sthOftrWeddingLunch_Mole.adaptType(ctClassLunch_A);
-		assertEquals("spoon.test.generics.testclasses.Mole", adapted.getQualifiedName());
-		//adapt A to scope of enclosing class of CelebrationLunch<Integer,Long,Double>.WeddingLunch<Mole>, which is CelebrationLunch<Integer,Long,Double>
-		adapted = sthOftrWeddingLunch_Mole.getEnclosingGenericTypeAdapter().adaptType(ctClassLunch_A);
-		assertEquals("java.lang.Double", adapted.getQualifiedName());
-		
+
+		// in disgust, the A of Lunch is bound to "Mole"
+		assertEquals("spoon.test.generics.testclasses.Mole", typingContextOfDisgust.adaptType(ctClassLunch_A).getQualifiedName());
+
+		// I don't understand the goal and utility of this one
+		assertEquals("java.lang.Double", typingContextOfDisgust.getEnclosingGenericTypeAdapter().adaptType(ctClassLunch_A).getQualifiedName());
```

Comment:
This shows that java class hierarchies of inner classes are different on each inner class level. In this test model there are two INDEPENDENT class hierarchies:
First is hierachy of top level classes:
```java
CelebrationLunch<K,L,M> extends Lunch<M,K>
```
and second there is hierarchy of inner class
```java
WeddingLunch<X> extends CelebrationLunch<Tacos, Paella, X> extends Lunch<X,Tacos>
```
Because in this wild example model, both ( 1. top level class `CelebrationLunch` and 2. inner class `WeddingLunch` ) extends from `Lunch<A,B>`, there is necessary to be able to adapt type parameters A and B in **both** contexts.

So this test line shows how to move from context of inner class `WeddingLunch` to the parent context of top level class `CelebrationLunch` and to adapt type parameters there.

May be it is confusing in this example that 
1) `WeddingLunch` is inner class of `CelebrationLunch`
2) `WeddingLunch` extends `CelebrationLunch` too
The `typingContextOfDisgust.getEnclosingGenericTypeAdapter()` is move from inner class to declaring class. It is NOT move from sub type to super type.